ding/league/shared_payoff.py

```diff
@@ -202,15 +202,20 @@ def _win_loss_reverse(result_: str, reverse_: bool) -> str:
             except Exception as e:
                 print("[ERROR] invalid job_info: {}\n\tError reason is: {}".format(job_info, e))
                 return False
-            key, reverse = self.get_key(home_id, away_id)
-            # Update with decay
-            for j in job_info_result:
-                for i in j:
-                    # All categories should decay
-                    self._data[key] *= self._decay
-                    self._data[key]['games'] += 1
-                    result = _win_loss_reverse(i, reverse)
-                    self._data[key][result] += 1
+            if home_id == away_id:  # self-play
+                key, reverse = self.get_key(home_id, away_id)
+                self._data[key]['draws'] += 1  # self-play defaults to draws
+                self._data[key]['games'] += 1
+            else:
+                key, reverse = self.get_key(home_id, away_id)
+                # Update with decay
+                for j in job_info_result:
+                    for i in j:
```

Comment:
Use meaningful or expressive variable name for `i` and `j`.

Reply:
ok, I will polish it